R/USA.R

```diff
@@ -0,0 +1,126 @@
+#' USA Class for downloading, cleaning and processing notification data
+#' @description Country specific information for downloading, cleaning
+#'  and processing covid-19 region data for USA.
+#'
+#' @details Inherits from `DataClass`
+#' @source https://raw.githubusercontent.com/nytimes/covid-19-data/master/us-states.csv # nolint
+#' @source https://raw.githubusercontent.com/nytimes/covid-19-data/master/us-counties.csv # nolint
+#' @export
+#' @examples
+#' \dontrun{
+#' region <- USA$new(verbose = TRUE, steps = TRUE, get = TRUE)
+#' region$return()
+#' }
+USA <- R6::R6Class("USA",
+  inherit = DataClass,
+  public = list(
+
+    # Core Attributes
+    #' @field country name of country to fetch data for
+    country = "United States of America (USA)",
+    #' @field supported_levels A list of supported levels.
+    supported_levels = list("1", "2"),
+    #' @field supported_region_names A list of region names in order of level.
+    supported_region_names = list(
+      "1" = "state",
+      "2" = "county"
+    ),
+    #' @field supported_region_codes A list of region codes in order of level.
+    supported_region_codes = list(
+      "1" = "iso_3166_2",
+      "2" = "fips"
+    ),
+    #' @field common_data_urls List of named links to raw data. Since there are
+    #'  no common data sources for level 1 and 2, this list is empty.
+    common_data_urls = list(
+      "1" = "https://raw.githubusercontent.com/nytimes/covid-19-data/master/us-states.csv", # nolint
```

Comment:
You can do this using level_data_urls and having common being empty. If that doesn't work it's a bug rather than a feature.

Reply:
merge was not allowing this with an empty list. I have moved them into `level_data_urls` and have `common_data_urls = list()` but had to change ```self$data_urls <- merge(
      self$common_data_urls,
      self$level_data_urls[[self$level]]
    )``` to ```self$data_urls <- unlist(
      list(self$common_data_urls, self$level_data_urls[[self$level]])
    )```

This works providing the common data urls is nested (e.g. `list("1" = list("1" = "url"), "2" = list("2" = "url"))`

This seems to work fine with the others and all tests pass